crates/compiler/test_gen/src/gen_list.rs

```diff
@@ -421,6 +421,133 @@ fn list_split() {
     );
 }
 
+#[test]
+#[cfg(any(feature = "gen-llvm", feature = "gen-wasm", feature = "gen-dev"))]
+fn list_split_on() {
+    assert_evals_to!(
+        r"
+        List.splitOn [] 1
+        ",
+        RocList::<RocList<i64>>::from_slice(&[RocList::<i64>::from_slice(&[])]),
+        RocList<RocList<i64>>
+    );
+    assert_evals_to!(
+        r"
+        List.splitOn [1] 1
+        ",
+        RocList::<RocList<i64>>::from_slice(&[
+            RocList::<i64>::from_slice(&[]),
+            RocList::<i64>::from_slice(&[]),
+        ]),
+        RocList<RocList<i64>>
+    );
+    assert_evals_to!(
+        r"
+        List.splitOn [1, 2, 3] 47
+        ",
+        RocList::<RocList<i64>>::from_slice(&[RocList::<i64>::from_slice(&[1, 2, 3])]),
+        RocList<RocList<i64>>
+    );
+    assert_evals_to!(
+        r"
+        List.splitOn [1, 2, 3, 4, 5] 3
+        ",
+        RocList::<RocList<i64>>::from_slice(&[
+            RocList::<i64>::from_slice(&[1, 2]),
+            RocList::<i64>::from_slice(&[4, 5]),
+        ]),
+        RocList<RocList<i64>>
+    );
+    assert_evals_to!(
+        r"
+        List.splitOn [1, 0, 1, 0, 1] 1
+        ",
+        RocList::<RocList<i64>>::from_slice(&[
+            RocList::<i64>::from_slice(&[]),
+            RocList::<i64>::from_slice(&[0]),
+            RocList::<i64>::from_slice(&[0]),
+            RocList::<i64>::from_slice(&[]),
+        ]),
+        RocList<RocList<i64>>
+    );
```

Comment:
I can see why it is surprising, but I think this is the right solution. This is was Str.split does currently and I don't think anyone has complained about it.
```roc
» Str.split "10101" "1"

["", "0", "0", ""] : List Str
```

Reply:
A more intuitive example might be splitting a file by newlines. If the first line of the file happens to only contain a new line, I definitely still want an empty string/empty list in my output, otherwise I would be losing relevant information about the file.